File: twoSum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int> &numbers, int target) {
        // Start typing your C/C++ solution below
        // DO NOT write int main() function
        map<int, int> numberSet;
        vector<int> result;
        for(int i = 0; i < numbers.size(); i++){
            // found
            if(numberSet.count(target - numbers[i]) != 0){
                result.push_back(numberSet[target - numbers[i]] + 1);
                result.push_back(i + 1);
            }
            else
                numberSet[numbers[i]] = i;
        }
        return result;
    }
};

File: twoSum.java
import java.util.*;

public static int[] twoSum(int[] numbers, int target) {
        HashMap numberSet = new HashMap();
        int[] result = new int[2];
        for(int i = 0; i < numbers.length; i++){
			// found
            if(!numberSet.isEmpty() && numberSet.containsKey(target - numbers[i])){
                result[0] = (Integer)numberSet.get(target - numbers[i]) + 1;
                result[1] = i + 1;
                System.out.println("found");
                return result;
            }
            else{
            	System.out.println(target - numbers[i]);
            	numberSet.put(numbers[i], i);
            }
            System.out.println(numberSet.size());
        }
        return result;
    }